lib/openstack/amqp/openstack_qpid_connection.rb

```diff
@@ -60,20 +64,13 @@ def session
 
   private
   def connection # :nodoc:
-    unless @connection
-      if @options.has_key?(:hostname) and @options.has_key?(:port)
-        @connection = create_connection
-      else
-        raise "Host and port required for OpenStack Qpid connection"
-      end
-    end
-    @connection
+    @connection ||= create_connection
   end
 
   def create_connection
-    url = "#{@options[:hostname]}:#{@options[:port]}"
-    opts = @options.select { |k, v| [:username, :password].include? k }
-    opts[:transport] = "ssl" if @options[:port] == 5671
-    Qpid::Messaging::Connection.new(:url => url, :options => opts)
+    url     = "#{@hostname}:#{@port}"
```

Comment:
This little bit of IPv6 violation will be fixed by #1824.  But, that PR will need to be rebased after this is merged.